mods/content/psionics/system/psionics/complexus/complexus_process.dm

```diff
@@ -70,6 +70,26 @@
 		to_chat(owner, SPAN_NOTICE("<b>Shift-left-click your Psi icon</b> on the bottom right to <b>view a summary of how to use them</b>, or <b>left click</b> it to <b>suppress or unsuppress</b> your psionics. Beware: overusing your gifts can have <b>deadly consequences</b>."))
 		to_chat(owner, "<hr>")
 
+//OFSTATION EDIT BEGIN
+/datum/psi_complexus/proc/adjust_psi_gen()
+
+	traumastam_regen = 0
+	var/mob/living/carbon/human/H = owner
+
+	if(H.get_shock() > 10)
+		traumastam_regen += 1
+	if(H.get_shock() > 50)
+		traumastam_regen += 1
+	if(H.reagents.has_reagent(/decl/material/liquid/hallucinogenics || /decl/material/liquid/psychoactives || /decl/material/liquid/psychotropics || /decl/material/liquid/narcotics))
+		traumastam_regen += 1.5
+	if(H.reagents.has_reagent(/decl/material/liquid/glowsap/gleam))
+		traumastam_regen += 2
+	else if(traumastam_regen == 0)
+		traumastam_regen -= 0.5
```

Comment:
Have these numbers been tested in-game?

Reply:
Yes. Normal stamina regen for psionics is a random 1-3, which given base psi stamina is 50, translated to 2-6% stamina per tick. With these numbers, the RNG is gone, but you do have to actually work for these gains. 10 shock is fairly easy, but 50 shock is a decent bit of pain to be in IMO, and half these drugs rn are honestly kind of stupid and make you basically unable to walk (which honestly we may want to tweak in the future). If it becomes a problem later we can nerf it but I don't think it'll be a huge issue, especially when you remember that psi stamina in combat is quickly drained by psionic armor.

As for gleam, its three eye, which gives you x-ray vision on top of this. Idk what we're gonna do with it, but anything that happens, I'm assuming it'll be rare.